### PR TITLE
Quickdraw Scabbards

### PR DESCRIPTION
**Pull request by assortedbeads**
## About The Pull Request

You can press the hotkey H to quickdraw some swords, much like you would a pistol.

I've ported over some code from Sojourn and edited it slightly to fit the vision I want.
This PR creates a new subtype of a holster called a scabbard. 
Like a holster it can be attached to your suit storage and a weapon can be stored into to be quickly accessed.
Unlike a holster the different scabbards only accept specific melee weapons (swords in this case), and only specific swords.

The types are;
- Scabbard: Can house all swords
- Makeshift Scabbard: Can house the Junkblade only

To make the first Scabbard you take a regular sheath, go to the crafting menu, click the recipe you want and it's turned into that sheath. It cannot be turned back.

The makeshift scabbard has a short recipe to craft it. I don't claim to be a sword expert I just looked at one irl for an hour irl, did a google search and tossed out something I thought was reasonable.

Sojourn also had some nice holster and scabbard sounds for drawing and sheathing weapons. 
From what I've seen they appear to not have any copyrights on them, so I've added them because audio feedback is always nice.
They were uploaded by [BigBadPossum](https://github.com/BigBadPossum)

Pictures below this, the icon sprites are reused sheath sprites we already had in use, with the exception of the makeshift one which is from Sojourn 
The mob sprites are from Sojourn. As far as I can tell they were by Possum again.
If you want to hear the sounds the files are here.
Sword1
Sword2
Sword3

## Why It's Good For The Game
Gives you another playstyle to follow. 
Don't want a pistol but want to have something you can quickly drop into a hand? Here you go.

As with all my PRs this is something I personally wanted, that I don't believe will make the game worse. Might even make it better.

Ultimately since it is a storage item it's a slight buff if you used these weapons frequently, since you have more space to work with. However, Eris is gun focused heavily and I've not seen anyone else going hard into full melee, I doubt this change will being a meta shift of any kind with how niche it is.

Additionally, these sword sheaths show up when you are examined. If you are wearing a void suit they do not but the sheath is clearly visible on your sprite. With the addition of unsheathing sounds, ideally people should be forewarned (as well as text chat) that you are drawing a weapon next to them.

It's a slight nerf to pistols since you can't just silently palm a pistol and blast someone. The chat log existed but now everyone can hear when you reach into your holster.

## Changelog
:cl:
add: Added 2 new scabbards and 2 equivalent recipes to obtain them
del: Removed old things
tweak: tweaked a few things
balance: rebalanced something
fix: fixed a few things
soundadd: added 4 new sounds, two for holster sheathing and unsheathing, two for scabbard sheathing and unsheathing
imageadd: added 4 new on-mob icons and 2 new icons
spellcheck: changed the grammar of some holster text to be less guncentric, since scabbards are a subtype of holsters.
code: ported code from Sojourn to allow holstering swords
/:cl:

### CODE REVIEW

**Comment by maikilangiolo** [comment]
It's like you can finally have a sword without sacrificing your belt or suit slot. The katana one would need imo a recolor/redesign to be more modern, since cyberpunkerino, and not identical to NT ones, but that's probably out of your competence..
NT gets fucked a bit, since the shortsword is weakest, while sabre, katana and junkblade are the highest hitters (with Moe katana being stupid good), so consider allowing the longsword, sword of truth and scourge too. 

This sheath looks more like a baldric, and those can fit a 40-inch blade.

**Comment by Humonitarian** [comment]
Good PR, but I think it'd be better if sheaths accepted all swords. Gameplay > realism.

Having multiple variants with different colours is a fun idea, though.

Finally, I would encourage considering a javeling holster for quickdrawing javelins.

**Comment by assortedbeads** [comment]
> It's like you can finally have a sword without sacrificing your belt or suit slot. The katana one would need imo a recolor/redesign to be more modern, since cyberpunkerino, and not identical to NT ones, but that's probably out of your competence.. NT gets fucked a bit, since the shortsword is weakest, while sabre, katana and junkblade are the highest hitters (with Moe katana being stupid good), so consider allowing the longsword, sword of truth and scourge too.
> 
> This sheath looks more like a baldric, and those can fit a 40-inch blade.

Yea I unfortunately have little to no spiriting ability yet. Working on it but currently unless I commission or someone offers, katana sprite currently out of reach.

You make a good point about the NT options, I've added in sheaths for the Longsword & Sword of Truth (Same sheath), and the Scourge.

**Comment by assortedbeads** [comment]
> Good PR, but I think it'd be better if sheaths accepted all swords. Gameplay > realism.
> 
> Having multiple variants with different colours is a fun idea, though.
> 
> Finally, I would encourage considering a javeling holster for quickdrawing javelins and spears.

I did consider an omni scabbard but current sheaths do act as those, and Im leaning more towards a "you've taken a universal sheath and customized it so you can quickly access your weapon" approach for these one. Or in the case of the junk blade cobbling one together since you know your scrap sword can take it.

The spear/javelin holster is a good idea. 
I wanted to initially have one for swords, one for spears and one for maces/hammers. 
However, since I also wanted them to telegraph when you have a sheath on and only sword sheath sprites were available, that's what I've focused the PR on for now.

I'm happy to expand it when I can port/get/make suitable sprites though.

**Comment by maikilangiolo** [comment]
The mace one can just be the baton holster but, well, with straps to tie it to the belt. 

Btw the claymore is the same size as the katana and junkblade. Honestly it feels weird to arbitrarily leave one?two? Swords out of the sheath. You sacrifice "no space" hidden pistol for a sword, and pistols hurt a lot without needing range. I doubt anyone will shit piss and cum you can have a suit sword.

You can't have both pistol holster and sword one, right? Then there's no problem with a "one size fits all"

**Comment by assortedbeads** [comment]
> You can't have both pistol holster and sword one, right? Then there's no problem with a "one size fits all"

You cant no, they're both utility accessories so it's one or the other.
Sheath accepts all swords with the latest commit.

File: code/modules/clothing/accessories/holster.dm [hunk 1]
```diff
@@ -130,3 +142,49 @@
...
+	can_hold = list(
+		/obj/item/tool/sword/nt/shortsword,
+		/obj/item/tool/sword/nt/longsword,
+		/obj/item/tool/sword/nt/scourge,
+		/obj/item/tool/sword/nt_sword,
+		/obj/item/tool/sword/saber,
+		/obj/item/tool/sword/katana,
+		/obj/item/tool/sword/katana/nano,
+		/obj/item/tool/sword,
+		/obj/item/tool/sword/improvised)
```
Comment (Firefox13) [inline code/modules/clothing/accessories/holster.dm:163]: I believe you can cut this all to just
`/obj/item/tool/sword`
I don't know how to do multi-line suggestions
But this is really cool, actually

**Review by TheShown911 — APPROVED** [review]
Yeah, this is nice.

File: code/modules/clothing/accessories/holster.dm (hunk 1 above)
Comment (assortedbeads) [reply code/modules/clothing/accessories/holster.dm]: Yea just did in the new commit, thanks for spotting that

File: code/modules/clothing/accessories/holster.dm [hunk 2]
```diff
@@ -8,13 +8,21 @@
...
 
 /obj/item/clothing/accessory/holster/proc/holster(obj/item/I, mob/living/user)
 	if(holstered && istype(user))
 		to_chat(user, SPAN_WARNING("There is already \a [holstered] holstered here!"))
 		return
 
-	if (!(I.slot_flags & SLOT_HOLSTER))
+	if (LAZYLEN(can_hold))
+		if(!is_type_in_list(I, can_hold))
+			to_chat(user, "<span class='warning'>[I] won't fit in [src]!</span>")
```
Comment (hyperioo) [inline code/modules/clothing/accessories/holster.dm:22]:
```suggestion
			to_chat(user, SPAN_WARNING("\The [I] won\'t fit in \the [src]!"))
```

**Pull request merged**